homeassistant/components/guardian/config_flow.py

```diff
@@ -90,15 +93,15 @@ async def async_step_user(self, user_input=None):
             title=info[CONF_UID], data={CONF_UID: info["uid"], **user_input}
         )
 
-    async def async_step_dhcp(self, discovery_info):
+    async def async_step_dhcp(self, discovery_info: dict) -> FlowResult:
```

Comment:
```suggestion
    async def async_step_dhcp(self, discovery_info: DiscoveryInfoType) -> FlowResult:
```
`from homeassistant.helpers.typing import DiscoveryInfoType`